Panic: invalid memory address or nil pointer dereference

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x951863e]

goroutine 579 [running]:
github.com/pingcap/tiflow/cdc/processor/sinkmanager.(*SinkManager).GetAllCurrentTableSpans(0x0)
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/processor/sinkmanager/manager.go:949 +0x7e
github.com/pingcap/tiflow/cdc/processor.(*processor).WriteDebugInfo(0xc0035d8a80, {0x7f53d25521f0, 0xc0034ce900})
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/processor/processor.go:928 +0x1d9
github.com/pingcap/tiflow/cdc/processor.(*managerImpl).writeDebugInfo(0xc00195aaa0, {0x7f53d25521f0, 0xc0034ce900})
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/processor/manager.go:363 +0x250
github.com/pingcap/tiflow/cdc/processor.(*managerImpl).handleCommand(0xc00195aaa0)
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/processor/manager.go:351 +0x205
github.com/pingcap/tiflow/cdc/processor.(*managerImpl).Tick(0xc00195aaa0, {0xcfe2ec0, 0xc003208280}, {0xcfc9810?, 0xc0014cb180?})
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/processor/manager.go:113 +0xd6
github.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run(0xc0034e7a20, {0xcfe2ec0?, 0xc003208280}, 0xc001960810, 0x2faf080, {0xadc7117, 0x9})
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/pkg/orchestrator/etcd_worker.go:290 +0x1e73
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker(0xc00340ed80, {0xcfe2ec0, 0xc003208280}, {0x7f53d250f190, 0xc00195aaa0}, 0xc0014cb180, 0x0?, {0xadc7117, 0x9})
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/capture/capture.go:614 +0x36c
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func4()
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/capture/capture.go:419 +0x98b
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/go/pkg/mod/golang.org/x/sync@v0.7.0/errgroup/errgroup.go:78 +0x98
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 525
	/go/pkg/mod/golang.org/x/sync@v0.7.0/errgroup/errgroup.go:75 +0x125
```

--- a/cdc/processor/processor.go
+++ b/cdc/processor/processor.go
@@ -924,6 +924,10 @@ func (p *processor) cleanupMetrics() {
 
 // WriteDebugInfo write the debug info to Writer
 func (p *processor) WriteDebugInfo(w io.Writer) error {
+	if !p.initialized.Load() {
+		fmt.Fprintln(w, "processor is not initialized")
+		return nil
+	}
 	fmt.Fprintf(w, "%+v\n%+v\n", *p.latestInfo, *p.latestStatus)
 	spans := p.sinkManager.r.GetAllCurrentTableSpans()
 	for _, span := range spans {

--- a/cdc/processor/processor_test.go
+++ b/cdc/processor/processor_test.go
@@ -18,6 +18,7 @@ import (
 	"encoding/json"
 	"fmt"
 	"math"
+	"os"
 	"sync/atomic"
 	"testing"
 
@@ -794,4 +795,11 @@ func TestProcessorDostNotStuckInInit(t *testing.T) {
 
 	require.Nil(t, p.Close())
 	tester.MustApplyPatches()
+}
+
+func TestProcessorNotInitialized(t *testing.T) {
+	globalVars, changefeedVars := vars.NewGlobalVarsAndChangefeedInfo4Test()
+	liveness := model.LivenessCaptureAlive
+	p, _, _ := initProcessor4Test(t, &liveness, false, globalVars, changefeedVars)
+	require.Nil(t, p.WriteDebugInfo(os.Stdout))
 }

